app/main.py

```diff
@@ -97,37 +99,70 @@ def grab_secrets(self):
+                _logger.info('Retrieving cert name:%s with version: %s output certFileName: %s', key_name, key_version, cert_filename)
                 cert = client.get_certificate(vault_base_url, key_name, key_version)
-                output_path = os.path.join(self._certs_output_folder, key_name)
+                output_path = os.path.join(self._certs_output_folder, cert_filename)
                 _logger.info('Dumping cert value to: %s', output_path)
                 with open(output_path, 'w') as cert_file:
                     cert_file.write(self._cert_to_pem(cert.cer))
 
-    def _dump_pfx(self, pfx, name):
+    def _dump_pfx(self, pfx, cert_filename, key_filename):
         import base64
         from OpenSSL import crypto
         p12 = crypto.load_pkcs12(base64.decodestring(pfx))
         pk = crypto.dump_privatekey(crypto.FILETYPE_PEM, p12.get_privatekey())
         cert = crypto.dump_certificate(crypto.FILETYPE_PEM, p12.get_certificate())
-        with open(os.path.join(self._keys_output_folder, name), 'w') as key_file:
-            key_file.write(pk)
-        with open(os.path.join(self._certs_output_folder, name), 'w') as cert_file:
-            cert_file.write(cert)
+
+        if (cert_filename == key_filename):
+            with open(os.path.join(self._keys_output_folder, key_filename), 'w') as key_file:                
+                _logger.info('Dumping key value to: %s', os.path.join(self._keys_output_folder, key_filename))
+                key_file.write(pk)
+
+            with open(os.path.join(self._certs_output_folder, cert_filename), 'w') as cert_file:
+                _logger.info('Dumping key value to: %s', os.path.join(self._certs_output_folder, cert_filename))
+                cert_file.write(cert)                
+        else:
+            # write to certs_keys folder when cert_filename and key_filename specified
+            with open(os.path.join(self._cert_keys_output_folder, key_filename), 'w') as key_file:
+                _logger.info('Dumping key value to: %s', os.path.join(self._keys_output_folder, key_filename))
+                key_file.write(pk)
+
+            with open(os.path.join(self._cert_keys_output_folder, cert_filename), 'w') as cert_file:
+                _logger.info('Dumping key value to: %s', os.path.join(self._cert_keys_output_folder, cert_filename))
+                cert_file.write(cert)    
+                    
+    @staticmethod
+    def _split_keyinfo(key_info):
+        key_parts = key_info.strip().split(':')
+        key_name = key_parts[0]
+        key_version = '' if len(key_parts) < 2 else key_parts[1]
+        cert_filename = key_name if len(key_parts) < 3 else key_parts[2]
+
+        # key_filename set to cert_filename when only cert_filename 
+        # key_filename default to key_name when cert and key filenames are not given
+        if (len(key_parts) == 4):
+            key_filename = key_parts[3]
+        elif (len(key_parts) == 3):
```

Comment:
this is not required, since `cert_filename` will default to `key_name` if `len(key_parts) < 3`

Reply:
This could be written as key_filename = cert_filename if len(key_parts) < 4 else key_parts[3]

This way if no special name is given at all, it gets the secret name, otherwise it gets the cert (or first name) and only if it has a 4th element does it get its own unique name.